torchmetrics/audio/sdr.py

```diff
@@ -0,0 +1,141 @@
+            filter coefficients instead of direct Gaussian elimination.
+            This can speed up the computation of the metrics in case the filters
+            are long. Using a value of 10 here has been shown to provide
+            good accuracy in most cases and is sufficient when using this
+            loss to train neural separation networks.
+        filter_length:
+            The length of the distortion filter allowed
+        zero_mean:
+            When set to True, the mean of all signals is subtracted prior to computation of the metrics
+        load_diag:
+            If provided, this small value is added to the diagonal coefficients of
+            the system metrics when solving for the filter coefficients.
+            This can help stabilize the metric in the case where some of the reference
+            signals may sometimes be zero
+        compute_on_step:
+            Forward only calls ``update()`` and returns None if this is set to False. default: True
+        dist_sync_on_step:
+            Synchronize metric state across processes at each ``forward()``
+            before returning the value at the step.
+        process_group:
+            Specify the process group on which synchronization is called. default: None (which selects the entire world)
+        dist_sync_fn:
+            Callback that performs the allgather operation on the metric state. When `None`, DDP
+            will be used to perform the allgather.
+
+    Raises:
+        ValueError:
+            If ``fast-bss-eval`` package is not installed
+
+    Example:
+        >>> from torchmetrics.audio import SDR
+        >>> import torch
+        >>> g = torch.manual_seed(1)
+        >>> preds = torch.randn(8000)
+        >>> target = torch.randn(8000)
+        >>> sdr = SDR()
+        >>> sdr(preds, target)
+        tensor(-12.0589)
+        >>> # use with pit
+        >>> from torchmetrics.audio import PIT
+        >>> from torchmetrics.functional.audio import sdr
+        >>> preds = torch.randn(4, 2, 8000)  # [batch, spk, time]
+        >>> target = torch.randn(4, 2, 8000)
+        >>> pit = PIT(sdr, 'max')
+        >>> pit(preds, target)
+        tensor(-11.6051)
+
+    .. note::
+       1. when pytorch<1.8.0, numpy will be used to calculate this metric, which causes ``sdr`` non-differentiable
+
+       2. using this metrics requires you to have ``fast-bss-eval`` install. Either install as ``pip install
+          torchmetrics[audio]`` or ``pip install fast-bss-eval``
+
+       3. preds and target need to have the same dtype, otherwise target will be converted to preds' dtype
+
+
+    References:
+        [1] Vincent, E., Gribonval, R., & Fevotte, C. (2006). Performance measurement in blind audio source separation.
+         IEEE Transactions on Audio, Speech and Language Processing, 14(4), 1462–1469.
+        [2] Scheibler, R. (2021). SDR -- Medium Rare with Fast Computations.
+        [3] https://github.com/fakufaku/fast_bss_eval
```

Comment:
This block is causing the issue. The docs compile fine when you remove this and the block in the functional/sdr.py one. 

Couldn't find why.